### ISSUE TITLE
Why `Option` instead of `Result`?

### ISSUE THREAD

**Issue by alexmozaidze** [DimChtz/rust-fstream#2]
I see that some functions may fail to do something, but the return type for them is `Option` rather than `Result`, why?

**Comment by DimChtz** [comment]
@alexmozaidze True, back when I made these functions I thought it would be better if the end user didn't have to worry if reading a file failed or not, and he should only care about getting the content (returning Option::none if fails or no content). However, I understand this could be bad because when you get `Option::none` you can't really know if the function read the file and the file was empty or if it simply failed. Any suggestions are welcomed !!
